nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-framework-core/src/main/java/org/apache/nifi/controller/FlowController.java

```diff
@@ -3846,6 +3851,73 @@ public QueueSize getTotalFlowFileCount(final ProcessGroup group) {
         return new QueueSize(count, contentSize);
     }
 
+    public class GroupStatusCounts {
```

Comment:
This is nit-picky but given that this class can be declared as static, since it doesn't reference FlowController private variables/methods, and given that FlowController is already nearly 4,000 lines of code, I'd recommend pulling this class out into its own .java file instead of making it an inner class of FlowController.

Reply:
And honestly, at that point, you could even get rid of the getGroupStatusCounts(ProcessGroup) method all together if you wanted, as you could just use the FlowController to get the root group and then just pass that root group to the constructor of GroupStatusCounts...

Reply:
This cannot be static as it needs access to the `processScheduler`

Reply:
Ah yes, I overlooked that. Is a bit difficult to find these things in a Github Diff page :)